frontend/src/components/assessment-creation/CreateAssessmentHeader.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+import {
+  Flex,
+  VStack,
+  Text,
+  Button,
+  Box,
+  Spacer,
+  HStack,
+  IconButton,
+} from "@chakra-ui/react";
+import {
+  TextOutlinedIcon,
+  SaveOutlinedIcon,
+  EyeOutlinedIcon,
+  MoreVerticalOutlineIcon,
+  ArrowBackOutlineIcon,
+} from "../../assets/icons";
+
+interface CreateAssessementHeaderProps {
+  assessmentName: string;
+  date: string;
+}
+
+const CreateAssessementHeader = ({
+  assessmentName,
+  date,
+}: CreateAssessementHeaderProps): React.ReactElement => {
+  return (
+    <Box
+      margin="-1.5em -2em 2em -2em"
+      padding="1.5em 2em 1.5em 2em"
+      borderBottom="1px"
+      borderBottomColor="grey.200"
+    >
+      <Flex minWidth="max-content">
+        <HStack spacing={6} alignItems="start">
+          <Button
+            size="sm"
+            color="blue.700"
+            leftIcon={<ArrowBackOutlineIcon />}
+          >
+            Back
+          </Button>
+          <VStack align="left">
+            <Text textStyle="subtitle1">
+              {assessmentName.length > 0
+                ? assessmentName
+                : "Untitled Assessment"}
+            </Text>
+            <Text textStyle="smallerParagraph"> Created {date}</Text>
+          </VStack>
+        </HStack>
+        <Spacer />
+        <HStack spacing={2}>
+          <HStack mr="2">
+            <EyeOutlinedIcon />
+            <Text color="blue.300" textStyle="link">
+              Preview
+            </Text>
+          </HStack>
+          <Button size="sm" variant="secondary" leftIcon={<SaveOutlinedIcon />}>
+            Save
+          </Button>
+          <Button size="sm" variant="primary" leftIcon={<TextOutlinedIcon />}>
+            Publish
+          </Button>
```

Comment:
I don't think you need to define a separate size here! I think if you just set the `minWidth` attribute here, you'll find that you can get the implementation to match the design. In general, I think the the `sm` size can be removed!